libcomps/src/python/src/CMakeLists.txt

```diff
@@ -44,13 +44,11 @@ endforeach(i)
 
 if (NOT SKBUILD)
     if (${pversion} EQUAL 2)
-        find_package (PythonLibs 2.6)
-        find_package (PythonInterp 2.6 REQUIRED)
+        find_package (PythonInterp 2 EXACT REQUIRED)
```

Comment:
I'd like to review this but I'm not sure how this works... however, in my recent attempt to build libdnf on rhel 7 (so with python 2), I had to drop the `EXACT` word from the line above to make it work, otherwise, it would error out. However, the `EXACT` word seems to be working fine with python 3...

Do you have any additional information on this @m-blaha?

Reply:
I think the `EXACT` keyword works as expected, it ensures that cmake only uses the exact specified version of python. What is causing the problems is the `find_package (PythonLibs)`, the versions of `PythonInterp` and `PythonLibs` have to match, so in your case its possible that by dropping the `EXACT` find_package found the same version for both.

I am not sure how to use `find_package (PythonLibs)` to find the some specified version, it says [here](https://cmake.org/cmake/help/v3.0/module/FindPythonLibs.html) you should use variables `Python_ADDITIONAL_VERSIONS`, `PYTHON_LIBRARY` and `PYTHON_INCLUDE_DIR`.

I would also like to know more about this. I don't understand how come it worked before with 3.7 or 3.8 python.

Reply:
I've just tested this on my RHEL 7.7 container and it's failing on the same error I described:
```
CMake Error at /usr/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:108 (message):
  Could NOT find PythonInterp: Found unsuitable version "2.7.5", but required
  is exact version "2" (found /bin/python2)
```
(the error is the same with or without `-DPYTHON_DESIRED:STRING=2`)

If I drop the `EXACT` from `find_package (PythonInterp 2 EXACT REQUIRED)`, it solves the problem. No idea if this is the right thing to do.

It still doesn't build though, it subsequently fails on:
```
CMake Error at src/python/src/CMakeLists.txt:99 (add_custom_target):
  add_custom_target cannot create target "pytest_run" because another target
  with the same name already exists.  The existing target is a custom target
  created in source directory
  "/home/lu/dev/libcomps/libcomps/src/python/src".  See documentation for
  policy CMP0002 for more details.
```

Reply:
I see, that is indeed a different issue. I tried it on F29 with both python 2 and 3 and I ran into the issue I described, it kind of looked similar from the outside.

If I remember correctly @m-blaha said that is was broken for python 2 even before. I also can't build it for python 2, not even without this PR. 
But since it fixes the problem for python 3.9 I would be inclined to merge it (It builds fine for me with  `-DPYTHON_DESIRED=3` for 3.7 and 3.8).

Reply:
@kontura hmm, I'm not sure what is the issue you've described? I don't see any obvious issue description in your previous comment :slightly_smiling_face: Also you talk about `PythonLibs`, but @m-blaha removed that check, so that's out as well...

Yeah, if it was broken for Python 2 before and it stays broken, I suppose we're not making things worse, but since we got into it, shall we fix it for Python 2 as well? As I said removing `EXACT` fixed it for me and the error message kind of complains the version doesn't match exactly, hinting at dropping the `EXACT`...

Anyway, I don't mind either, feel free to merge if you want :slightly_smiling_face:

Reply:
I only used similar solution to one we already have in libdnf. In fact I did not investigate problems with python2, I just mentioned it is broken.

Reply:
@lukash I just meant the issue of different versions between `PythonInterp` and `PythonLibs`. Ok I am going to merge it. Python 2 can be resolved later if need be.